src/DataStructures/Tensor/Expressions/Evaluate.hpp

```diff
@@ -8,35 +8,115 @@
+ * the two that is used to determine the (potentially reordered) ordering of the
+ * elements of the desired LHS Tensor`s ::Symmetry and typelist of
+ * \ref SpacetimeIndex "TensorIndexType"s.
+ *
+ * @tparam RhsTensorIndexList the typelist of TensorIndex of the RHS
+ * TensorExpression, e.g. `ti_a_t`, `ti_b_t`, `ti_c_t`
+ * @tparam LhsTensorIndexList the typelist of TensorIndexs of the desired LHS
+ * tensor, e.g. `ti_b_t`, `ti_c_t`, `ti_a_t`
+ * @tparam RhsSymmetry the ::Symmetry of the RHS indices
+ * @tparam RhsTensorIndexTypeList the RHS TensorExpression's typelist of
+ * \ref SpacetimeIndex "TensorIndexType"s
+ */
+template <typename RhsTensorIndexList, typename LhsTensorIndexList,
+          typename RhsSymmetry, typename RhsTensorIndexTypeList,
+          size_t NumIndices = tmpl::size<RhsSymmetry>::value,
+          typename IndexSequence = std::make_index_sequence<NumIndices>>
+struct LhsTensor;
+
+template <typename RhsTensorIndexList, typename... LhsTensorIndices,
+          typename RhsSymmetry, typename RhsTensorIndexTypeList,
+          size_t NumIndices, size_t... Ints>
+struct LhsTensor<RhsTensorIndexList, tmpl::list<LhsTensorIndices...>,
+                 RhsSymmetry, RhsTensorIndexTypeList, NumIndices,
+                 std::index_sequence<Ints...>> {
+  static constexpr std::array<size_t, NumIndices> lhs_tensorindex_values = {
+      {LhsTensorIndices::value...}};
+  static constexpr std::array<size_t, NumIndices> rhs_tensorindex_values = {
+      {tmpl::at_c<RhsTensorIndexList, Ints>::value...}};
+  static constexpr std::array<size_t, NumIndices> lhs_to_rhs_map = {
+      {std::distance(
+          rhs_tensorindex_values.begin(),
+          alg::find(rhs_tensorindex_values, lhs_tensorindex_values[Ints]))...}};
+
+  // Desired LHS Tensor's Symmetry and typelist of TensorIndexTypes
+  using symmetry =
+      Symmetry<tmpl::at_c<RhsSymmetry, lhs_to_rhs_map[Ints]>::value...>;
+  using tensorindextype_list =
+      tmpl::list<tmpl::at_c<RhsTensorIndexTypeList, lhs_to_rhs_map[Ints]>...>;
+};
 
 /*!
  * \ingroup TensorExpressionsGroup
- * \brief Evaluate a Tensor Expression with LHS indices set in the template
- * parameters
+ * \brief Evaluate a right hand side tensor expression to a tensor with the LHS
+ * index order set in the template parameters
  *
- * @tparam LhsIndices the indices on the left hand side of the tensor expression
- * @return Tensor<typename T::type, typename T::symmetry, typename
- * T::index_list>
+ * \details Uses the right hand side (RHS) TensorExpression's index ordering
+ * (`T::args_list`) and the desired left hand side (LHS) tensor's index ordering
+ * (`LhsTensorIndices`) to construct a LHS Tensor with that LHS index ordering.
+ * This can carry out the evaluation of a RHS tensor expression to a LHS tensor
+ * with the same index ordering, such as \f$L_{ab} = R_{ab}\f$, or different
+ * ordering, such as \f$L_{ba} = R_{ab}\f$.
+ *
+ * ### Usage
```

Comment:
I suggest "Usage" -> "Usage example" or "Example use" to make it more evident that other kinds of tensors and tensor expressions work too

Reply:
Done!